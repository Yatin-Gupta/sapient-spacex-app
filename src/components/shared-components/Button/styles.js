/**
 * Button Stylesheet in JSS
 */
const buttonColor = "#c5e09b";
const buttonHoverColor = "#a6d05a";
const buttonActiveColor = "#7cba01";

const styles = {
  button: {
    backgroundColor: buttonColor,
    borderRadius: "5px",
    width: "80px",
    padding: "10px",
    textAlign: "center",
    border: "0px",
    fontWeight: "600",
    "&:hover": {
      cursor: "pointer",
      backgroundColor: buttonHoverColor,
    },
  },
  activeButton: {
    backgroundColor: buttonActiveColor,
  },
};

export default styles;
