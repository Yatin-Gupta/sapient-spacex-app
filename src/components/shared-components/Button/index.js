import React from "react";
import { createUseStyles } from "react-jss";
import styles from "./styles";

const useStyles = createUseStyles(styles);

/**
 * Button Component
 * @param {*} props
 */
function Button(props) {
  const classes = useStyles();
  const { isactive, children, className } = props;
  let buttonClasses = classes.button;
  // Set class name by active status
  buttonClasses = isactive ? `${buttonClasses} ${classes.activeButton}` : buttonClasses;
  buttonClasses = className ? `${buttonClasses} ${className}` : buttonClasses;
  return (
    <React.Fragment>
      <button onClick={props.onClick} className={buttonClasses} aria-pressed={isactive ? "true" : "false"}>
        {children}
      </button>
    </React.Fragment>
  );
}

export default Button;
