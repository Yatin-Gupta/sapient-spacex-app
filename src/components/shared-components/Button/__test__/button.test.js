import React from "react";
import ReactDOM from "react-dom";
import Button from "../index";

import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";

afterEach(cleanup);

it("Button component renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<Button>Click Me</Button>, div);
  ReactDOM.unmountComponentAtNode(div);
});

it("Button component renders with isactive status passed as props", () => {
  const { getByTestId } = render(<Button isactive={true}></Button>);
  expect(getByTestId("button")).toHaveAttribute("aria-pressed", "true");
});

it("Button component matches the snapshot", () => {
  const tree = renderer.create(<Button></Button>).toJSON();
  expect(tree).toMatchSnapshot();
});
