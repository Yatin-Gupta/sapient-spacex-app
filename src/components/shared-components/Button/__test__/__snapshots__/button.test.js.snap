// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component matches the snapshot 1`] = `
<button
  aria-pressed="false"
  className="button-0-2-1"
  data-testid="button"
/>
`;
